src/neo-vm/ExecutionEngine.cs

```diff
@@ -333,6 +339,13 @@ private bool ExecuteInstruction()
                             context.EvaluationStack.Push(context.AltStack.Pop());
                             break;
                         }
+                    case OpCode.ISNULL:
+                        {
+                            bool b = context.EvaluationStack.Peek().IsNull;
```

Comment:
Should be a `Peek` or `Pop`?

Reply:
`Peek` is right I think.

Reply:
Should be `Peek()`.

Reply:
I don't think Peek is a good idea... we don't have any other of this kind, do we?
Why not keep the pattern DUP ISNULL ? Why not change NUMEQUAL to the same new pattern, not consuming the item?

Reply:
It do consume the item. Please check the code carefully.

Reply:
oh,sorry! I just saw the Set operation being used... is it better for performance? Ok then, one issue less :joy: